src/AzureAuth/Commands/CommandInfo.cs

```diff
@@ -21,13 +22,13 @@ public class CommandInfo
         /// This method executes the info process.
         /// </summary>
         /// <param name="logger">The logger.</param>
-        /// <param name="fileSystem">The file system.</param>
+        /// <param name="telemetryService">The telemetry service.</param>
         /// <returns>The error code: 0 is normal execution, and the rest means errors during execution.</returns>
-        public int OnExecute(ILogger<CommandInfo> logger, IFileSystem fileSystem)
+        public int OnExecute(ILogger<CommandInfo> logger, ITelemetryService telemetryService)
         {
             Assembly assembly = Assembly.GetExecutingAssembly();
             string azureauthVersion = assembly.GetName().Version.ToString();
-            string deviceID = TelemetryDeviceID.GetAsync(fileSystem).Result;
+            string deviceID = (telemetryService as TelemetryServiceBase)?.MachineGUID;

```

Comment:
I'm confused why we need to do this. Casting the telemetry service here couples it to a particular instance, We should be able to continue using the TelemetryDeviceID interface as is, no?